R/motifs.R

```diff
@@ -402,3 +402,57 @@ triad_census <- function(graph) {
     graph = graph
   )
 }
+
+
+#' Find motifs with a callback function
+#'
+#' This function searches for motifs in a graph and calls a user-provided
+#' callback function for each motif found.
+#'
+#' @param graph Graph object, the input graph.
+#' @param size The size of the motif, currently sizes 3 and 4 are supported in
+#'   directed graphs and sizes 3-6 in undirected graphs.
+#' @param cut_prob Numeric vector giving the probabilities that the search
+#'   graph is cut at a certain level. Its length should be the same as the size
+#'   of the motif (the `size` argument).
+#'   If `NULL`, the default, no cuts are made.
+#' @param callback A function to call for each motif found. The function should
+#'   accept two arguments: `vids` (integer vector of vertex IDs in the motif,
+#'   1-based indexing) and `isoclass` (the isomorphism class of the motif, 1-based).
+#'   The function should return `TRUE` to continue the search or `FALSE` to stop it.
+#' @return `NULL`, invisibly. This function is called for its side effects (calling
+#'   the callback function for each motif).
+#' @seealso [motifs()], [count_motifs()], [isomorphism_class()]
+#'
+#' @export
+#' @family graph motifs
+#' @cdocs igraph_motifs_randesu_callback
+#'
+#' @examples
```

Comment:
I am not convinced by this example. I think we need an example that shows you can *end* the search after finding a given number of motifs. Before knowing that I didn't see the point of the function.